ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/TypeContext.java

```diff
@@ -0,0 +1,58 @@
+package io.confluent.ksql.execution.codegen;
+
+import io.confluent.ksql.schema.ksql.types.SqlType;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TypeContext {
+  private SqlType sqlType;
+  private final List<SqlType> inputTypes = new ArrayList<>();
+  private final Map<String, SqlType> lambdaTypeMapping = new HashMap<>();
+
+  public SqlType getSqlType() {
+    return sqlType;
+  }
+
+  public void setSqlType(final SqlType sqlType) {
+    this.sqlType = sqlType;
+  }
+
+  public List<SqlType> getInputTypes() {
+    if (inputTypes.size() == 0) {
+      return null;
```

Comment:
nit: when public APIs can return `null`, it's often preferable to wrap it in an `Optional` so that any method callee doesn't forget to handle it

alternatively, is there any reason not to just return the empty list?

Reply:
There used to be a reason for the null, but there isn't anymore. Switched it to just return the list